### PR TITLE
New backend option

### PR DESCRIPTION
**Pull request by timwis**
I found out today from Socrata that `?nbe=true` can be used to create the dataset in the new backend. They advised that this should be used for (a) geospatial datasets, and (b) huge tabular datasets (15-20 million rows). For everything else, the old backend should be used for now, since the new backend doesn't support everything just yet (ie. export formats).

I also removed that whitespace on empty lines that I created in my last PR. My bad.

Note that I imagine in a few months this won't be necessary as all dataset creation will automatically be in the new backend.

### CODE REVIEW

File: sodapy/__init__.py [hunk 1]
```diff
@@ -84,22 +84,23 @@ def create(self, name, **kwargs):
             category : must exist in /admin/metadata
             tags : array of tag strings
```
Comment (xmunoz) [inline sodapy/__init__.py]: Not from this commit, but can you change the verbiage here to the more pythonic-sounding "list of tag strings"?

File: sodapy/__init__.py [hunk 2]
```diff
@@ -84,22 +84,23 @@ def create(self, name, **kwargs):
...
             row_identifier : field name of primary key
+            new_backend : whether to create the dataset in the new backend
         '''
-        public = kwargs.pop("public", False)
-        published = kwargs.pop("published", False)
-        
+        new_backend = kwargs.pop("new_backend", False)
+        resource = "/api/views.json" + ("?nbe=true" if new_backend else "")
```
Comment (xmunoz) [inline sodapy/__init__.py]: For readability, can you instead do something like this?
```
resource = "/api/views.json"
if new_backend:
    resource +=  "?nbe=true"
```

**Comment by xmunoz** [comment]
Looks good, just some minor comments. Also, can you add the description that you just provided here about when to use the new backend to the `README`?

**Comment by chrismetcalf** [comment]
To be more consistent with [the new API versioning scheme](https://dev.socrata.com/changelog/2015/11/25/soda-api-versioning.html), is it worth at least calling it something like `version=2.1` or something?

**Comment by xmunoz** [comment]
Hmm....after reading this, I think that the version of the api you are hitting should be called in the constructor, and not in any particular method:

```
client = Socrata("opendata.socrata.com", "SuperSecretToken", api_version=2.1)
```

Maybe default this keyward arg to 2.0 for now (with the appropriate readme documentation so that power users can take advantage of it), and when @chrismetcalf gives the green light, change the default to 2.1. What do you think?

**Comment by timwis** [comment]
Thanks for the feedback @xmunoz. Hmm, `api_version` in the constructor would definitely make sense if there were a chance of other methods doing something differently (there may be). Does it make sense if `create` is the only method affected?

More broadly, the idea of passing `2.1` for the param value instead of `True`/`False` - I'd worry about being _too_ specific with the version. For instance, what about when `2.2` comes out before the new backend is default? I imagine we'd want to use a `> 2.0` conditional to test for whether or not to use `?nbe=true`. I feel like I'm not articulating it well, but `?nbe=true` seems like a different thing than an API version.

Having said all that, I'm _totally_ fine with both of these suggestions and would be happy to implement them. Just wanted to toss out another perspective. Let me know what you guys think and I'll get typing.

**Comment by xmunoz** [comment]
That's a legitimate concern, I withdrawal my initial proposal and propose more concerns :)

From my understanding, it looks like other methods are also affected since [new Soql functions](https://dev.socrata.com/docs/functions/) are supported in v. 2.1, as well as new datatypes and content types. This would affect `get`, `upsert` and `update`. Though to support all of this new functionality would be much larger undertaking that you probably initially intended with this PR.

The way this library is now though, I'm not sure it could successfully upload data in a geojson format. The only use-case this library might work for (with the PR as is) is huge tabular datasets, but I haven't tried yet, so I don't know. It might [timeout](https://github.com/xmunoz/sodapy/blob/c15c0117be935d62a55c867dc74fe63df4c1b183/sodapy/__init__.py#L232) before finishing...

Basically, I'm worried that if I merge this PR then one could create a dataset in the new backend, but not much else.

**Comment by xmunoz** [comment]
Ok, after more thought, I think I've come up with a good solution. The user probably doesn't need to know about things like when to use this backend or that, so in the interest of usability, a good approach might just be to add some logic that asses what type of data it's dealing with and use the appropriate backend based on that. This way, the library just does the right thing without any special inputs from the user. I'm going to steal your PR on a PR idea and give this a go.

**Comment by timwis** [comment]
> The way this library is now though, I'm not sure it could successfully upload data in a geojson format. The only use-case this library might work for (with the PR as is) is huge tabular datasets, but I haven't tried yet, so I don't know. It might timeout before finishing...
> 
> Basically, I'm worried that if I merge this PR then one could create a dataset in the new backend, but not much else.
You lost me here - I don't quite understand

**Comment by xmunoz** [comment]
Ok, so there are other reasons to use the new backend, got it. What I was trying to get at before was the the use-cases that you outlined (geojson, large datasets) aren't actually supported by this library yet. If someone were trying to upload geojson data to their newly created dataset in the new backend, it would fail. Since there are other reasons you might was to use it though, I'm fine with that.

Ultimately, I'd like to work towards a more general solution to the problem of updates happening faster to the Socrata API than they can be added here, but this makes sense for now.
